### PR TITLE
Backup the webserver

### PR DESCRIPTION
**Pull request by stephdl**
Backup of the module, we backup 

```
# volumes
volumes/sftpgo_backups
volumes/sftpgo_config
volumes/websites
# backup data
state/databases
state/sftpgo.conf.d/API_KEY
# we only backup custom, conf are expanded
state/conf.d/*.custom
state/php*-fpm-custom.d/*.custom
```

password of sftpgo are well saved
Configuration in state saved
configuration well expanded
container are started after restoration (sftpgo, nfinx,phpfpm*)

### CODE REVIEW

File: imageroot/actions/restore-module/60systemd [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+# expand configuration
+subprocess.run(["systemctl", "--user", "enable","--now", "nginx.service"])
+subprocess.run(["systemctl", "--user", "enable","--now", "sftpgo.service"])
+subprocess.run(["systemctl", "--user", "restart", "nginx.service"])
+subprocess.run(["systemctl", "--user", "restart", "sftpgo.service"])
+
+# detect if a service has configurations, hence start it
+PhpServiceArray = glob.glob('php*-fpm-custom.d')
```
Comment (gsanchietti) [inline imageroot/actions/restore-module/60systemd:35]: Do we really need to use variables with uppercase fist letter? 
Still, it's just a matter of code style :)

**Review by gsanchietti — COMMENTED** [review]
It looks good to me, but I'd prefer a review by @DavidePrincipi which already saw the module code

File: imageroot/actions/restore-module/50traefik [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+default_traefik_id = agent.resolve_agent_id('traefik@node')
+if default_traefik_id is None:
+    sys.exit(2)
+
+for file in glob.iglob("databases/vhosts/*.ini"):
+    config = configparser.ConfigParser()
+    config.read(file)
+    ServerNames = config['vhost']['ServerNames'].split(',')
+    http2https = True if config['vhost']['http2https'] == 'enabled' else False
+    lets_encrypt = True if config['vhost']['lets_encrypt'] == 'enabled' else False
```
Comment (DavidePrincipi) [inline imageroot/actions/restore-module/50traefik:41]: The `==` operator already returns a boolean value. I think the expression is simpler and more readable also in this way:

```suggestion
    http2https = config['vhost']['http2https'] == 'enabled'
    lets_encrypt = config['vhost']['lets_encrypt'] == 'enabled'
```

Also a wrapping  `bool()` could be added...

This suggestion applies also to other similar expressions in the following lines

**Review by DavidePrincipi — APPROVED** [review]
Thank you, it seems good! ...only a little coding remark

File: imageroot/actions/restore-module/50traefik (hunk 2 above)
Comment (stephdl) [reply imageroot/actions/restore-module/50traefik:41]: done

**Pull request merged**